File: hb-notification/src/main/java/com/ntnt/highblog/notification/model/dto/response/NotificationRes.java
package com.ntnt.highblog.notification.model.dto.response;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.ntnt.highblog.notification.enums.NotificationType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@JsonInclude(JsonInclude.Include.NON_NULL)
public class NotificationRes {

    private Long id;
    private Long senderId;
    private Long sourceId;
    private String content;
    private NotificationType type;

    @JsonProperty("sender")
    private NotificationSenderRes senderRes;
}

File: high-blog-webapp/src/models/response/NotificationRes.ts
import {UserRes} from "./UserRes";
export enum NotificationType {
    POST = "POST",
    COMMENT = "COMMENT",
    SUBSCRIPTION = "SUBSCRIPTION",
    ADMIN = "ADMIN",
}

export interface NotificationRes{
    id: number,
    senderId: number,
    sourceId: number,
    content: string,
    type: NotificationType,
    sender: UserRes,
};
